WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchBatchLoggerService.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GeeksCoreLibrary.Core.Models;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using MySqlConnector;
+using Newtonsoft.Json;
+using WiserTaskScheduler.Core.Models;
+using WiserTaskScheduler.Modules.Branches.Interfaces;
+using WiserTaskScheduler.Modules.Branches.Models;
+
+namespace WiserTaskScheduler.Modules.Branches.Services;
+
+/// <inheritdoc cref="IBranchBatchLoggerService" />
+public class BranchBatchLoggerService(IOptions<WtsSettings> wtsOptions, ILogger<BranchBatchLoggerService> logger) : BackgroundService, IBranchBatchLoggerService
+{
+    private readonly ConcurrentQueue<BranchMergeLogModel> mergeLogQueue = new();
+    private readonly BatchLoggerOptions options = wtsOptions.Value.BatchLoggerOptions;
+
+    /// <inheritdoc />
+    public void LogMergeAction(BranchMergeLogModel logData)
+    {
+        mergeLogQueue.Enqueue(logData);
+    }
+
+    /// <inheritdoc />
+    public override async Task StopAsync(CancellationToken cancellationToken)
+    {
+        // Override StopAsync to wait for the log queue to be completely flushed before shutdown.
+        logger.LogInformation("BatchLogger is stopping. Flushing remaining logs.");
```

Comment:
Is there a specific reason the LogService of the WTS is not used?

Reply:
Because I forgot it exists. But I also don't have any log settings for this background service. I made something so that the `BranchQueueService` passes it's logger settings to this one. Can you check if you agree with this approach? I couldn't think of a better way.

Reply:
The logger settings could be added to the `BatchLoggerOptions`. If not provided it can still use the one from the branch queue service since it is part of that. For now we will leave it with this so that the PR can be approved. This would be a possible future improvement is we notice this needs to be managed separately from the branch queue service.